datafusion/physical-plan/src/repartition/mod.rs

```diff
@@ -685,9 +690,12 @@ impl RepartitionExec {
         partitioning: Partitioning,
         metrics: RepartitionMetrics,
         context: Arc<TaskContext>,
+        preserve_order: bool,
     ) -> Result<()> {
-        let mut partitioner =
-            BatchPartitioner::try_new(partitioning, metrics.repartition_time.clone())?;
+        let mut partitioner = BatchPartitioner::try_new(
+            partitioning.clone(),
+            metrics.repartition_time.clone(),
+        )?;
```

Comment:
The clones seem unnecessary.
```suggestion
        let mut partitioner = BatchPartitioner::try_new(
            partitioning,
            metrics.repartition_time,
        )?;
```